### PR TITLE
Update fslex and fsyacc to 11.0.1

### LINKED ISSUE #14583: Update FsLexYacc
Hello @vzarytovskii and @KevinRansom, I would like to update the `fslex` and `fsyacc` build tools.
The main reason for updating these tools is that the fix of https://github.com/fsprojects/FsLexYacc/pull/154 recently got published to NuGet over at https://github.com/fsprojects/FsLexYacc.

From what I understood the compiler takes a source dependency on `FsLexYacc` instead of using the NuGet package. What would need to be done to update this?

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #14583 
I copied to source from the [11.0.1](https://www.nuget.org/packages/FsLexYacc/11.0.1)

### CODE REVIEW

**Comment by vzarytovskii** [comment]
That's a lot of changes @nojaf do you happen to know if I can easily find commits list in the fslexyacc repo, which were included in the release?

**Comment by vzarytovskii** [comment]
@dsyme was fslexyacc changed in any way when having a local copy of compiler? Because in this case, just updating it from upstream might not work properly.

**Comment by nojaf** [comment]
Do you have any idea what release you were on, to begin with?
There are some release notes [here](https://github.com/fsprojects/FsLexYacc/blob/master/RELEASE_NOTES.md).

All the CI passed (expect for something which I believe to be unrelated, https://github.com/dotnet/fsharp/runs/10576589137)

**Comment by vzarytovskii** [comment]
Yep, apparently one of the last updates was in https://github.com/dotnet/fsharp/pull/6355 to 8.0.1

**Comment by nojaf** [comment]
That would explain why there are so many changes I guess.

File: buildtools/fslex/Lexing.fs [hunk 1]
```diff
@@ -1,423 +1,456 @@
 // (c) Microsoft Corporation 2005-2009.
+module FSharp.Text.Lexing
 
 #nowarn "47" // recursive initialization of LexBuffer
 
+open System.Collections.Generic
 
-#if INTERNALIZED_FSLEXYACC_RUNTIME
```
Comment (dsyme) [inline buildtools/fslex/Lexing.fs:6]: It looks like we expect `INTERNALIZED_FSLEXYACC_RUNTIME` to be an option over the source, given we define
```
    <DefineConstants>INTERNALIZED_FSLEXYACC_RUNTIME;$(DefineConstants)</DefineConstants>
```

Comment (dsyme) [reply buildtools/fslex/Lexing.fs:6]: However we only use this in fslex and fsyacc projects.  Maybe it doesn't matter

**Comment by dsyme** [comment]
I looked through and didn't see any changes in fslex or fsyacc that affect how the grammar is being processed. By the look of it I would expect exactly the same tables and generated source to be generated

**Comment by nojaf** [comment]
@dsyme @vzarytovskii is there anything left to address here?

**Review by T-Gro — APPROVED** [review]

File: buildtools/fslex/Lexing.fs (hunk 1 above)
Comment (vzarytovskii) [reply buildtools/fslex/Lexing.fs:6]: It was factored out in https://github.com/fsprojects/FsLexYacc/pull/144

**Review by vzarytovskii — APPROVED** [review]

**Comment by dsyme** [comment]
For me this is ok

**Pull request merged**